### ISSUE TITLE
Get all catalogs/collections available from STAC endpoint

### ISSUE THREAD

**Issue by SethDocherty** [easierdata/ipfs-stac#37]
The methods `searchSTACByBox` and `searchSTACByBoxIndex` require users to pass a list of collection IDs. Before using either of these methods, I will need to know the collections id's as they are exactly spelled.  If the user is not knowledgeable of the collections from a given STAC endpoint, they will need to either:

1. Browse to the endpoint to manually identify the collection ids.
2. Import `pystac` and `pystac_client`, create a new catalog object, and generate the list, which is outside the scope of the `Web3` object instantiated from `ipfs-stac`

We should add a new property to `Web3` that stores either a list of collection id's or a list of `Collection` pystac object. My first hunch is that we should go the latter route since the collection object could be helpful with the wealth of details it contains.

```Python
from pystac_client import Client

catalog = Client.open("stac.easierdata.info")

# NOTE: calling this method returns a generator object so it needs to be encapsulated as a list to get anything meaningful
list(catalog.get_all_collections())
[<CollectionClient id=landsat-c2l1>, <CollectionClient id=GEDI_L4A_AGB_Density_V2_1_2056.v2.1>]
```

**Comment by SethDocherty** [comment]
Note, we could use `catalog.describe` but must take into consideration the **gotcha** described [here](https://stacspec.org/en/tutorials/1-read-stac-python/#Explore-the-High-Level-Catalog-Information).

**Comment by SethDocherty** [comment]
This has been resolved with the #46
